src/pages/Content/index.js

```diff
@@ -84,247 +42,311 @@ async function getUserInfo() {
       console.log('Data has been sent to the server');
     })
-    .catch(() => {
-      console.log('Internal server error');
+    .catch((err) => {
+      console.log('Internal server error in updateSites:', err);
     });
 
-  axios.post("http://localhost:4000/api/websites/addSite",{
-    _id: url
+  axios
+    .post('http://localhost:4000/api/websites/addSite', {
+      _id: currentURL,
     })
     .then((response) => {
       console.log(response);
     })
-    .catch(() => {
-      console.log('Internal server error');
+    .catch((err) => {
+      console.log('Internal server error in addSite:', err);
     });
-    createQuestionnaire(userInfo.id, url, hostname);
 
-  //Highlight everything
-  paragraphs = document.getElementsByTagName('p');
-  var i = 0
-  for (const paragraph of paragraphs) {
-    //console.log(paragraph.textContent)
-    if (first) {
-      colors.push([paragraph.style['background-color'], getRandomColor()]);
+  axios.get("http://localhost:4000/api/websites/getUserHighlights", {params: {
+    url: currentURL,
+    userID: currentUserInfo.id
+  }}).then((res) => {
+    for(highlightObj in res) {
+      console.log("Highlight from DB:", highlightObj.highlightSelection)
+      highlightText('#ffc107', highlightObj.highlightSelection);
     }
-    paragraph.style['background-color'] = colors[i][1];
-    i++;
+  }).catch((err) => {
+    console.log('Internal server error in getUserHighlights:', err);
+  });
+
+  if (first) {
+    createQuestionnaire(currentUserInfo.id, currentURL, currentHostname);
+
+    //Highlight everything
+    even = (even + 1) % 2;
+
+    let tooltip = document.createElement('span');
+    tooltip.className = 'tool_tip';
+
+    document.body.appendChild(tooltip);
+
+    let isToolTipVisible = false;
+    let lastSelection = null;
+    let lastSelectionObj = null;
+    var markerEl = null;
+
+    render(<ToolComponent>aa</ToolComponent>, tooltip);
+    tooltip.style.position = 'absolute';
+    tooltip.style.visibility = 'hidden';
+    tooltip.style.display = 'none';
+
+    const renderToolTip = (mouseX, mouseY, selection) => {
+      mouseX = mouseX - 50;
+      mouseY = mouseY - 40;
+      if (selection.length < 60) {
+        mouseX = mouseX - 20;
+      }
+      tooltip.style.top = mouseY + 'px';
+      tooltip.style.left = mouseX + 'px';
+      tooltip.style.visibility = 'visible';
+      tooltip.style.display = 'block';
+      isToolTipVisible = true;
+    };
+
+    const closeToolTip = () => {
+      tooltip.style.visibility = 'hidden';
+      tooltip.style.display = 'none';
+      isToolTipVisible = false;
+      clearSelection();
+    };
+
+    function clearSelection() {
+      if (window.getSelection) {window.getSelection().removeAllRanges();}
+      else if (document.selection) {document.selection.empty();}
+    }
+
+    var startX = 0;
+    var endX = 0;
+    var startY = 0;
+    var endY = 0;
+
+    //Close the tool tip
+    document.addEventListener('mousedown', (e) => {
```

Comment:
### ISSUE 1

 Need more testing on more pages to see if the tooltip properly disappears
  You can double click to "remove" a selection, but the tooltip is still visible need to workout cases
        
Albeit this one may be a very simple fix
Share findings if anyone can.